hypershift-operator/controllers/nodepool/metrics/metrics.go

```diff
@@ -193,11 +202,80 @@ func createFailureConditionToNodePoolsCountMap(knownConditionToExpectedStatus ma
 	return &res
 }
 
+const (
+	unexpectedAWSOutputErrorReason          = "unexpected AWS output"
+	failedToCallAWSErrorReason              = "failed to call AWS"
+	unableToMarshalPricingDataErrorReason   = "unable to marshal pricing data"
+	unableToUnMarshalPricingDataErrorReason = "unable to un-marshal pricing data"
+	noErroReason                            = ""
+)
+
 type vCpusDetail struct {
 	vCpusCount            int32
 	vCpusCountErrorReason string // set only if vCpusCount == -1
 }
 
+func extractCPUFromInstanceTypeName(instanceTypeName string, ec2Client ec2iface.EC2API, pricingClient pricingiface.PricingAPI) (int64, string) {
+	// First try to get it through EC2 DescribeInstanceTypes API
+	ec2InstanceTypes, err := ec2Client.DescribeInstanceTypes(&ec2.DescribeInstanceTypesInput{
```

Comment:
Shouldn't the CPU count be read from some config map first?
Idea would be to be able to handle `instanceTypeName` for which both calls to AWS fail. WDYT?